packer/install-dependencies.sh

```diff
@@ -47,17 +47,16 @@ cat << EOF >> ${RELEASE_NOTES_FILEPATH}
   - zip
 EOF
 
+
+MOBY_VERSION="3.0.4"
+installMoby
+echo "  - moby v${MOBY_VERSION}" >> ${RELEASE_NOTES_FILEPATH}
+
 if [[ ${FEATURE_FLAGS} == *"docker-engine"* ]]; then
-    DOCKER_ENGINE_REPO="https://apt.dockerproject.org/repo"
-    installDockerEngine
     overrideDockerEngineStorageDriver
     echo "  - docker-engine v${DOCKER_ENGINE_VERSION}" >> ${RELEASE_NOTES_FILEPATH}
     installGPUDrivers
```

Comment:
Why are we only installing GPU drivers with the docker-engine feature flag if this PR is to "use moby for GPU instances"?

Reply:
Seems like this was a bad assumption before.
Is there some env var that we can look at for GPU instances?

Reply:
This the script that we run in the pipeline that builds our VHDs, it doesn't run at provisioning. Packer runs the script to install a bunch of software on a VM. There is no notion of GPU/not GPU instance at that stage. The env var that you are looking for that tells if a VM is a GPU node at provisioning is in kubernetescustomscript.sh

Reply:
So we definitely don't want to install GPU drivers unless it's a GPU enabled node as this changes how containers are executed.
I guess we could modify the installation to only install the updated `/etc/docker/daemon.json` (which is what modifies the container runtime to use nvidia-docker's runtime) at at provision time.

What do you think? Any other ideas?

Reply:
There are two different things happening:

1) kubernetesinstalls: downloads all necessary software, this is called by both the VHD packer script (install-dependencies.sh) and CSE for VMs not using the VHD image (base Ubuntu image)
2) kubernetesconfigs: runtime configurations + uninstall software that isn't needed for a particular VM instance

We only run `ensureGPUDrivers`  https://github.com/Azure/aks-engine/blob/master/parts/k8s/kubernetesconfigs.sh#L420 on GPU VMs at provisioning. 

We should pull all the GPU drivers in the VHD and only enable the nvidia-runtime if the node is a GPU instance at provisioning. Also, we should uninstall the GPU drivers if the node is NOT a GPU instance. Currently, we don't do this because we have two different VHD flavors: Moby for non-GPU and docker-engine for GPU nodes. In the future, both GPU and non-GPU nodes will build from the same Moby VHD so we need to add extra uninstalling of the nvidia-drivers in CSE when the node doesn't need them.

Keep in mind that all of this also needs to work for clusters that opt out of the VHD and build on top of the Vanilla Ubuntu image (this is a requirement for sovereign clouds that don't support our base image). So the full GPU flow needs to be functional installing when everything during CSE.